### PR TITLE
Allow state-label-element to show attributes

### PR DESCRIPTION
**Pull request by guillempages**
Add an additional (optional) parameter to the state-label element,
so that instead of showing the state, an attribute can be shown instead.

Can be used to e.g. add a climate entity to the floorplan,
but show the current temperature instead of the desired one.

## Proposed change

Add the possibility to set "attribute" as a parameter to the picture-elements state-label type,
so that instead of showing the state, a specific attribute can be shown instead, but the actions
still refer to the original entity.

Can be used to e.g. show the current temperature instead of the desired one in a label, but still be able to control it with a simple click.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
title: My Awesome Home
views:
  - title: Example
    cards:
      - type: picture-elements
        image: /local/floorplan.png
        title: My Home
        elements:
          - type: state-label
            entity: sun.sun
            prefix: "Sun elevation:"
            attribute: elevation
            style:
              top: 15%
              left: 70%
              color: white
          - type: state-label
            entity: binary_sensor.updater
            attribute: unexisting_attribute
            style:
              top: 30%
              left: 20%
              color: white
          - type: state-label
            entity: sensor.not_existing
            style:
              top: 30%
              left: 70%
              color: white

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/12151

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 1]
```diff
@@ -58,6 +59,25 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
...
+            "entity",
+            this._config.entity
+          )}
+        ></hui-warning-element>
+      `;
+    }
+
+    const stateDisplay = stateObj
+      ? computeStateDisplay(this.hass.localize, stateObj, this.hass.language)
+      : "-";
```
Comment (zsarnett) [inline src/panels/lovelace/elements/hui-state-label-element.ts]: Not sure I see why you are computing the display here? Just inline it in the HTML template

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 2]
```diff
@@ -70,13 +90,9 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
...
       >
-        ${this._config.prefix}${stateObj
-          ? computeStateDisplay(
-              this.hass.localize,
-              stateObj,
-              this.hass.language
-            )
-          : "-"}${this._config.suffix}
+        ${this._config.prefix}${attribute
+          ? attributeDisplay
```
Comment (zsarnett) [inline src/panels/lovelace/elements/hui-state-label-element.ts]: Best practice for ifs/ternary is for the condition that is expected to be true most of the time should be in the if section while the rare condition in the else section. So for this I would do ! attribute then state else attribute

File: src/panels/lovelace/elements/hui-state-label-element.ts (hunk 1 above)
Comment (guillempages) [reply src/panels/lovelace/elements/hui-state-label-element.ts]: I didn't want nested ternaries (they are hard to read), so I pre-computed the values here

Comment (zsarnett) [reply src/panels/lovelace/elements/hui-state-label-element.ts]: hmm maybe I am missing something but at this point in the code (and even the old code) is stateObj ever undefined? We have already checked for it in the above IF and return if it is undefined

Comment (iantrich) [reply src/panels/lovelace/elements/hui-state-label-element.ts]: You're not. stateObj is already checked

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 3]
```diff
@@ -58,6 +59,25 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
...
+            this._config.entity
+          )}
+        ></hui-warning-element>
+      `;
+    }
+
+    const stateDisplay = stateObj
+      ? computeStateDisplay(this.hass.localize, stateObj, this.hass.language)
+      : "-";
+    const attributeDisplay = stateObj ? stateObj.attributes[attribute] : "-";
```
Comment (iantrich) [inline src/panels/lovelace/elements/hui-state-label-element.ts]: remove this. stateObj exists

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 4]
```diff
@@ -70,13 +90,9 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
...
-        ${this._config.prefix}${stateObj
-          ? computeStateDisplay(
-              this.hass.localize,
-              stateObj,
-              this.hass.language
-            )
-          : "-"}${this._config.suffix}
+        ${this._config.prefix}${!attribute
+          ? stateDisplay
+          : attributeDisplay}${this._config.suffix}
```
Comment (iantrich) [inline src/panels/lovelace/elements/hui-state-label-element.ts]:
```suggestion
          : stateObj.attributes[attribute]}${this._config.suffix}
```

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 5]
```diff
@@ -70,13 +90,9 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
...
       >
-        ${this._config.prefix}${stateObj
-          ? computeStateDisplay(
-              this.hass.localize,
-              stateObj,
-              this.hass.language
-            )
-          : "-"}${this._config.suffix}
+        ${this._config.prefix}${!attribute
+          ? stateDisplay
```
Comment (iantrich) [inline src/panels/lovelace/elements/hui-state-label-element.ts]: change this back to computeStateDisplay

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 6]
```diff
@@ -45,6 +45,7 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
     }
 
     const stateObj = this.hass.states[this._config.entity!];
+    const attribute = this._config.attribute!;
```
Comment (iantrich) [inline src/panels/lovelace/elements/hui-state-label-element.ts]: remove this and just reference it as needed in full

File: src/panels/lovelace/elements/hui-state-label-element.ts [hunk 7]
```diff
@@ -58,6 +59,25 @@ class HuiStateLabelElement extends LitElement implements LovelaceElement {
       `;
     }
 
+    if (attribute && !stateObj.attributes[attribute]) {
```
Comment (iantrich) [inline src/panels/lovelace/elements/hui-state-label-element.ts]:
```suggestion
    if (this._config.attribute&& !stateObj.attributes[attribute]) {
```

File: src/panels/lovelace/elements/hui-state-label-element.ts (hunk 3 above)
Comment (iantrich) [reply src/panels/lovelace/elements/hui-state-label-element.ts]: I see why you did check as it was in the existing code, but that was unnecessary

**Review by bramkragten — APPROVED** [review]

**Comment by guillempages** [comment]
Oops. sorry, I actually did the documentation PR, but forgot to add the link :-(
I've updated the description.
https://github.com/home-assistant/home-assistant.io/pull/12151

**Pull request merged**